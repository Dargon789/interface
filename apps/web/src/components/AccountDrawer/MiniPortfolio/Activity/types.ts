import { Currency } from '@uniswap/sdk-core'
import { GraphQLApi } from '@universe/api'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import {
  TransactionOptions,
  TransactionStatus,
  UniswapXOrderDetails,
} from 'uniswap/src/features/transactions/types/transactionDetails'

/**
 * TODO: refactor parsing / Activity so that all Activity Types can have a detail sheet.
 */

export type Activity = {
  id: string
  hash?: string
  chainId: UniverseChainId
  outputChainId?: UniverseChainId
  status: TransactionStatus
  offchainOrderDetails?: UniswapXOrderDetails
  statusMessage?: string
  timestamp: number
  title: string
  descriptor?: string | JSX.Element
  logos?: Array<string | undefined>
  // TODO(WEB-3839): replace Currency with CurrencyInfo
  currencies?: Array<Currency | undefined>
  otherAccount?: string
  from: string
  options?: TransactionOptions
  prefixIconSrc?: string
  suffixIconSrc?: string
  isSpam?: boolean
  type?: GraphQLApi.TransactionType
}

export type ActivityMap = { [id: string]: Activity | undefined }
