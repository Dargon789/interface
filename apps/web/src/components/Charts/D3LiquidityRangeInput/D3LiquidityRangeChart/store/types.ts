import { GraphQLApi } from '@universe/api'
import { TickAlignment } from 'components/Charts/D3LiquidityRangeInput/D3LiquidityRangeChart/utils/priceToY'
import { ChartEntry } from 'components/Charts/LiquidityRangeInput/types'
import { PriceChartData } from 'components/Charts/PriceChart'
import * as d3 from 'd3'
import { UseSporeColorsReturn } from 'ui/src/hooks/useSporeColors'

export type ChartState = {
  dimensions: {
    width: number
    height: number
  }
  dragCurrentTick?: ChartEntry
  dragCurrentY?: number
  dragStartTick?: ChartEntry
  dragStartY: number | null
  dynamicZoomMin: number
  hoveredTick?: ChartEntry
  hoveredY?: number
  initialViewSet: boolean
  isChartHovered?: boolean
  isFullRange: boolean
  maxPrice?: number
  minPrice?: number
  panY: number
  selectedHistoryDuration: GraphQLApi.HistoryDuration
  selectedPriceStrategy?: DefaultPriceStrategy
  zoomLevel: number
}

export type AnimationParams = {
  targetZoom: number
  targetPan: number
  targetMinPrice?: number
  targetMaxPrice?: number
  duration?: number
}

export type RenderingContext = {
  colors: UseSporeColorsReturn
  dimensions: {
    width: number
    height: number
  }
  priceData: PriceChartData[]
  liquidityData: ChartEntry[]
  tickScale: ((tick: string) => number) & {
    domain: () => string[]
    bandwidth: () => number
    range: () => [number, number]
  }
  priceToY: ({ price, tickAlignment }: { price: number; tickAlignment?: TickAlignment }) => number
  yToPrice: (y: number) => number
}

export enum DefaultPriceStrategy {
  STABLE = 'stable',
  WIDE = 'wide',
  ONE_SIDED_UPPER = 'one_sided_upper',
  ONE_SIDED_LOWER = 'one_sided_lower',
}

export interface Renderer {
  draw(): void
}

type Renderers = {
  priceLineRenderer: Renderer | null
  liquidityBarsRenderer: Renderer | null
  liquidityRangeAreaRenderer: Renderer | null
  minMaxPriceLineRenderer: Renderer | null
  scrollbarContainerRenderer: Renderer | null
  minMaxPriceIndicatorsRenderer: Renderer | null
  currentTickRenderer: Renderer | null
  liquidityBarsOverlayRenderer: Renderer | null
  timescaleRenderer: Renderer | null
}

export type ChartActions = {
  setChartState: (state: Partial<ChartState>) => void
  setPriceStrategy: (strategy: DefaultPriceStrategy) => void
  setTimePeriod: (timePeriod: GraphQLApi.HistoryDuration) => void
  updateDimensions: (dimensions: { width: number; height: number }) => void
  handlePriceChange: (changeType: 'min' | 'max', price?: number) => void
  initializeView: (params?: { minPrice: number | null; maxPrice: number | null }) => void
  initializeRenderers: ({
    g,
    timescaleG,
    context,
  }: {
    g: d3.Selection<SVGGElement, unknown, null, undefined>
    timescaleG: d3.Selection<SVGGElement, unknown, null, undefined>
    context: RenderingContext
  }) => void
  createHandleDragBehavior: (lineType: 'min' | 'max') => d3.DragBehavior<any, unknown, unknown>
  createTickBasedDragBehavior: () => d3.DragBehavior<any, unknown, unknown>
  centerRange: () => void
  zoom: (targetZoom: number) => void
  zoomIn: () => void
  zoomOut: () => void
  reset: (params?: { animate?: boolean; minPrice?: number; maxPrice?: number }) => void
  drawAll: () => void
  animateToState: (params: AnimationParams) => void
  incrementMax: () => void
  decrementMax: () => void
  incrementMin: () => void
  decrementMin: () => void
}

export type ChartStoreState = ChartState & {
  renderers: Renderers
  renderingContext: RenderingContext | null
  dynamicZoomMin: number
  actions: ChartActions
}
