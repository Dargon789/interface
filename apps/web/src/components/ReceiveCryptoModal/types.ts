import { FORServiceProvider } from 'uniswap/src/features/fiatOnRamp/types'

export enum ReceiveModalState {
  DEFAULT = 0,
  CEX_TRANSFER = 1,
  QR_CODE = 2,
  CEX_TRANSFER_CHOOSE_PLATFORM = 3,
}

export type ReceiveCryptoModalInitialState =
  | {
      modalState: Exclude<ReceiveModalState, ReceiveModalState.QR_CODE | ReceiveModalState.CEX_TRANSFER_CHOOSE_PLATFORM>
    }
  | { modalState: ReceiveModalState.QR_CODE; qrCodeAddress: string }
  | { modalState: ReceiveModalState.CEX_TRANSFER_CHOOSE_PLATFORM; serviceProvider: FORServiceProvider }
