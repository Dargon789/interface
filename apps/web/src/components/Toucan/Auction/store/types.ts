import { UniverseChainId } from 'uniswap/src/features/chains/types'

// potentially missing clearing price
export interface AuctionDetails {
  auctionId: string
  chainId: UniverseChainId
  tokenSymbol: string
  tokenAddress: string
  tokenName: string // this is missing from what Rob sent
  logoUrl: string // this is missing from what Rob sent
  creatorAddress: string
  startBlock: number
  endBlock: number
  totalSupply: string
  tickSize: string
  graduationThreshold: number
  bidTokenAddress: string
}

export interface CheckpointData {
  clearingPrice: string
  cumulativeMps: number
}

export enum DisplayMode {
  VALUATION = 'VALUATION',
  TOKEN_PRICE = 'TOKEN_PRICE',
}

interface AuctionState {
  auctionDetails: AuctionDetails | null
  checkpointData: CheckpointData | null
  tokenColor?: string
  displayMode: DisplayMode
}

interface AuctionActions {
  setTokenColor: (color?: string) => void
  setDisplayMode: (mode: DisplayMode) => void
}

export type AuctionStoreState = AuctionState & {
  actions: AuctionActions
}
