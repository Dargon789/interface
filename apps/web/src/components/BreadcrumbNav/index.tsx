import { Currency } from '@uniswap/sdk-core'
import { useState } from 'react'
import { useTranslation } from 'react-i18next'
import { Link } from 'react-router'
import { CopyHelper } from 'theme/components/CopyHelper'
import { Flex, styled, Text, TextProps, useMedia } from 'ui/src'
import { TestID } from 'uniswap/src/test/fixtures/testIDs'
import { shortenAddress } from 'utilities/src/addresses'

export const BreadcrumbNavContainer = styled(Flex, {
  row: true,
  alignItems: 'center',
  gap: '$gap4',
  mb: 20,
  width: 'fit-content',
})

export const BreadcrumbNavLink = ({ to, children, ...rest }: { to: string; children: React.ReactNode } & TextProps) => {
  return (
    <Link to={to} style={{ textDecoration: 'none' }}>
      <Text
        display="flex"
        alignItems="center"
        animation="fast"
        color="$neutral2"
        $platform-web={{
          textDecoration: 'none',
        }}
        hoverStyle={{ color: '$neutral2Hovered' }}
        {...rest}
      >
        {children}
      </Text>
    </Link>
  )
}

const CurrentPageBreadcrumbContainer = styled(Flex, {
  row: true,
  gap: 6,
})

// This must be an h1 to match the SEO title, and must be the first heading tag in code.
const PageTitleText = styled(Text, {
  tag: 'h1',
  fontWeight: 'inherit',
  fontSize: 'inherit',
  lineHeight: 'inherit',
  color: '$neutral1',
  whiteSpace: 'nowrap',
  margin: 0,
})

// Used in both TDP & PDP.
// On TDP, currency is defined & poolName is undefined. On PDP, currency is undefined & poolName is defined.
export const CurrentPageBreadcrumb = ({
  address,
  currency,
  poolName,
}: {
  address: string
  currency?: Currency
  poolName?: string
}) => {
  const { t } = useTranslation()
  const isNative = currency?.isNative
  const tokenSymbolName = currency?.symbol ?? t('tdp.symbolNotFound')

  const media = useMedia()
  const shouldEnableCopy = !media.md
  const [isBreadcrumbHover, setIsBreadcrumbHover] = useState(false)

  return (
    <CurrentPageBreadcrumbContainer
      aria-current="page"
      data-testid="current-breadcrumb"
      onMouseEnter={() => setIsBreadcrumbHover(true)}
      onMouseLeave={() => setIsBreadcrumbHover(false)}
    >
      <PageTitleText>{currency ? tokenSymbolName : poolName}</PageTitleText>
      {(!currency || !isNative) && (
        <CopyHelper
          toCopy={address}
          iconPosition="right"
          iconSize={16}
          iconColor="$neutral2"
          color="$neutral2"
          disabled={!shouldEnableCopy}
          externalHover={isBreadcrumbHover}
          dataTestId={TestID.BreadcrumbHoverCopy}
        >
          {shortenAddress({ address })}
        </CopyHelper>
      )}
    </CurrentPageBreadcrumbContainer>
  )
}
