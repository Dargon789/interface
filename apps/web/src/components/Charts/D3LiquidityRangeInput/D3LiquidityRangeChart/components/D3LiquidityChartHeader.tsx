import { Currency, Price } from '@uniswap/sdk-core'
import { DisplayCurrentPrice } from 'components/Liquidity/DisplayCurrentPrice'
import { Flex } from 'ui/src'
import { SegmentedControl, SegmentedControlOption } from 'ui/src/components/SegmentedControl/SegmentedControl'

export function D3LiquidityChartHeader({
  price,
  isLoading,
  creatingPoolOrPair,
  currencyControlOptions,
  baseCurrency,
  handleSelectToken,
}: {
  price?: Price<Currency, Currency>
  isLoading?: boolean
  creatingPoolOrPair?: boolean
  currencyControlOptions: SegmentedControlOption<string>[]
  baseCurrency: Currency
  handleSelectToken: (option: string) => void
}) {
  return (
    <Flex
      row
      justifyContent="space-between"
      alignItems="center"
      p="$padding16"
      $sm={{ row: false, alignItems: 'flex-start', gap: '$gap8' }}
    >
      <DisplayCurrentPrice price={price} isLoading={isLoading} />
      {!creatingPoolOrPair && (
        <SegmentedControl
          options={currencyControlOptions}
          selectedOption={baseCurrency.symbol ?? ''}
          onSelectOption={handleSelectToken}
          size="smallThumbnail"
        />
      )}
    </Flex>
  )
}
