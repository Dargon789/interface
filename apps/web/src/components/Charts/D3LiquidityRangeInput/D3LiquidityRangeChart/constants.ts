const RANGE_INDICATOR_WIDTH = 16
const LIQUIDITY_CHART_HEIGHT = 200
const LIQUIDITY_CHART_PADDING = 14

// Chart Dimension Constants
export const CHART_DIMENSIONS = {
  // Chart dimensions
  LIQUIDITY_CHART_HEIGHT,
  LIQUIDITY_CHART_WIDTH: 120,
  TIMESCALE_HEIGHT: 20,
  RANGE_INDICATOR_WIDTH,
  LIQUIDITY_SECTION_OFFSET: RANGE_INDICATOR_WIDTH + 1,
  LIQUIDITY_BARS_SPACING: 10,
  LIQUIDITY_BAR_HEIGHT: 3,
  LIQUIDITY_BAR_SPACING: 1,
  SOLID_MIN_MAX_LINE_HEIGHT: 3,
  RANGE_INDICATOR_MIN_HEIGHT: 40,
  TRANSPARENT_MIN_MAX_LINE_HEIGHT: 30,
  PRICE_DOT_RADIUS: 4,
  // Chart total height used for loading state
  LIQUIDITY_CHART_TOTAL_HEIGHT: LIQUIDITY_CHART_HEIGHT + LIQUIDITY_CHART_PADDING,

  // Interactive element sizes
  DRAG_BOUNDARY_MARGIN: 10,
} as const

// Chart behavior constants
export const CHART_BEHAVIOR = {
  // zoom
  ZOOM_MIN: 0.01, // fallback, overridden by calculateDynamicZoomMin
  ZOOM_MAX: 3,
  ZOOM_FACTOR: 1.3,

  // pinch
  PINCH_DELTA_SCALE: 0.02,
  PINCH_ZOOM_FACTOR_MIN: 0.5,
  PINCH_ZOOM_FACTOR_MAX: 2.0,

  // animation
  ANIMATION_DURATION: 400,
} as const
