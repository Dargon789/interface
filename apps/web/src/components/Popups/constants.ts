// Do not change, this is Sonner's default max width
export const POPUP_MAX_WIDTH = 348
