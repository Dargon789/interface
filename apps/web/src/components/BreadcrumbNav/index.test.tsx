import userEvent from '@testing-library/user-event'
import { CurrentPageBreadcrumb } from 'components/BreadcrumbNav'
import { NATIVE_CHAIN_ID } from 'constants/tokens'
import { TokenFromList } from 'state/lists/tokenFromList'
import { render, screen } from 'test-utils/render'
import { nativeOnChain } from 'uniswap/src/constants/tokens'
import { UniverseChainId } from 'uniswap/src/features/chains/types'
import { TestID } from 'uniswap/src/test/fixtures/testIDs'

describe('BreadcrumbNav', () => {
  it('renders hover components correctly', async () => {
    const currency = new TokenFromList({
      chainId: 1,
      address: '0x2260fac5e5542a773aa44fbcfedf7c193bc2c599',
      name: 'Wrapped BTC',
      decimals: 18,
      symbol: 'WBTC',
    })
    const { asFragment } = render(
      <CurrentPageBreadcrumb address="0x2260fac5e5542a773aa44fbcfedf7c193bc2c599" currency={currency} />,
    )
    expect(asFragment()).toMatchSnapshot()

    await userEvent.hover(screen.getByTestId('current-breadcrumb'))
    expect(screen.getByTestId(TestID.BreadcrumbHoverCopy)).toBeInTheDocument()
    await userEvent.unhover(screen.getByTestId('current-breadcrumb'))
    expect(screen.queryByTestId(TestID.BreadcrumbHoverCopy)).not.toBeInTheDocument()
  })

  it('does not display address hover for native tokens', async () => {
    const ETH = nativeOnChain(UniverseChainId.Mainnet)
    const { asFragment } = render(<CurrentPageBreadcrumb address={NATIVE_CHAIN_ID} currency={ETH} />)
    expect(asFragment()).toMatchSnapshot()

    await userEvent.hover(screen.getByTestId('current-breadcrumb'))
    expect(screen.queryByTestId(TestID.BreadcrumbHoverCopy)).not.toBeInTheDocument()
  })
})
