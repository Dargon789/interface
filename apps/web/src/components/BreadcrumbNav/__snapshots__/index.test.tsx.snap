// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`BreadcrumbNav > does not display address hover for native tokens 1`] = `
<DocumentFragment>
  <span
    class=""
    style="display: contents;"
  >
    <div
      aria-current="page"
      class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _gap-6px"
      data-testid="current-breadcrumb"
    >
      <h1
        class="font_body _display-inline _boxSizing-border-box _color-neutral1 _fontFamily-f-family _wordWrap-break-word _fontSize-inherit _lineHeight-inherit _fontWeight-inherit _whiteSpace-nowrap _mt-0px _mr-0px _mb-0px _ml-0px"
        data-disable-theme="true"
      >
        ETH
      </h1>
    </div>
    <div
      style="display: contents;"
    />
  </span>
</DocumentFragment>
`;

exports[`BreadcrumbNav > renders hover components correctly 1`] = `
<DocumentFragment>
  <span
    class=""
    style="display: contents;"
  >
    <div
      aria-current="page"
      class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _gap-6px"
      data-testid="current-breadcrumb"
    >
      <h1
        class="font_body _display-inline _boxSizing-border-box _color-neutral1 _fontFamily-f-family _wordWrap-break-word _fontSize-inherit _lineHeight-inherit _fontWeight-inherit _whiteSpace-nowrap _mt-0px _mr-0px _mb-0px _ml-0px"
        data-disable-theme="true"
      >
        WBTC
      </h1>
      <div
        class="_color-_platformweb_neutral2 _opacity-0hover-0--8 _opacity-0active-0--6 _display-flex _flexBasis-auto _boxSizing-border-box _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-row _gap-4px _cursor-pointer _position-relative _alignItems-center"
        style="transition: 100ms;"
      >
        <div
          class="_whiteSpace-_platformweb_nowrap _overflowX-_platformweb_hidden _overflowY-_platformweb_hidden _textOverflow-_platformweb_ellipsis _display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _maxWidth-calc100-0px38074892"
        >
          0x2260...C599
        </div>
        <div
          class="_clear-_platformweb_both _display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _position-relative _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column"
        />
        <div
          class="_display-flex _alignItems-stretch _flexBasis-auto _boxSizing-border-box _minHeight-0px _minWidth-0px _flexShrink-0 _flexDirection-column _position-relative _width-16px _height-16px"
        />
      </div>
    </div>
    <div
      style="display: contents;"
    />
  </span>
</DocumentFragment>
`;
